### PR TITLE
TST:Add test to check single category col returns series with single row slice

### LINKED ISSUE #14011: DataFrame with one column, which is categorical: single rows convert to scalars
#### Code Sample, a copy-pastable example if possible

``` python
In [2]: pd.DataFrame({'x': pd.Categorical('a b c d e'.split())}).iloc[0]
Out[2]: 'a'
```
#### Expected Output

I think this should be a Series with one element:

``` python
0    a
dtype: category
Categories (5, object): [a, b, c, d, e]
```

by analogy with the non-Categorical case:

``` python
In [3]: pd.DataFrame({'x': 'a b c d e'.split()}).iloc[0]
Out[3]:
x    a
Name: 0, dtype: object
```

This bug also exhibits with `.loc[0]` and `.ix[0]`. It doesn't happen if the original dataframe has more than one column, even if they're all categorical.
## INSTALLED VERSIONS

commit: None
python: 2.7.11.final.0
python-bits: 64
OS: Darwin
OS-release: 15.5.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_GB.UTF-8

pandas: 0.18.1
nose: 1.3.7
pip: 8.1.2
setuptools: 18.5
Cython: 0.23.4
numpy: 1.11.1
scipy: 0.18.0
statsmodels: 0.6.1
xarray: None
IPython: 4.0.1
sphinx: 1.3.1
patsy: 0.4.1
dateutil: 2.5.3
pytz: 2016.6.1
blosc: None
bottleneck: 1.0.0
tables: 3.2.2
numexpr: 2.6.0
matplotlib: 1.5.1
openpyxl: 2.2.6
xlrd: 0.9.4
xlwt: 1.0.0
xlsxwriter: 0.7.7
lxml: 3.4.4
bs4: 4.4.1
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.9
pymysql: None
psycopg2: None
jinja2: 2.8
boto: 2.40.0
pandas_datareader: None

### PR DESCRIPTION
**Pull request by ryankarlos**
- [x] closes #14011
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/frame/test_dtypes.py [hunk 1]
```diff
@@ -762,6 +762,12 @@ def test_astype_categoricaldtype_class_raises(self, cls):
         with pytest.raises(TypeError, match=xpr):
             df["A"].astype(cls)
 
+    def test_singlerow_slice_categoricaldtype_gives_series(self):
+        df = pd.DataFrame({"x": pd.Categorical("a b c d e".split())})
```
Comment (mroeschke) [inline pandas/tests/frame/test_dtypes.py]: Can you add the issue number?

File: pandas/tests/frame/test_dtypes.py [hunk 2]
```diff
@@ -762,6 +762,12 @@ def test_astype_categoricaldtype_class_raises(self, cls):
...
+        raw_cat = pd.Categorical(["a"], categories=["a", "b", "c", "d", "e"])
+        expected = pd.Series(raw_cat, index=["x"], name=0, dtype="category")
+        tm.assert_series_equal(df.iloc[0], expected)
```
Comment (mroeschke) [inline pandas/tests/frame/test_dtypes.py]: Can you use `tm.assert_series_equal(result, expected)`

Comment (gfyoung) [reply pandas/tests/frame/test_dtypes.py]: To clarify @mroeschke comment, he means this:

~~~python
result = df.iloc[0]
tm.assert_series_equal(df.iloc[0], expected)
~~~

And while we're at it, let's also add a new line beneath `expected` to make it easier to read.

Comment (ryankarlos) [reply pandas/tests/frame/test_dtypes.py]: Done

File: pandas/tests/frame/test_dtypes.py (hunk 1 above)
Comment (ryankarlos) [reply pandas/tests/frame/test_dtypes.py]: Done

**Review by mroeschke — APPROVED** [review]

**Comment by gfyoung** [comment]
Thanks @ryankarlos !

**Pull request merged**